lib/ansible/cli/doc.py

```diff
@@ -729,7 +731,12 @@ def _get_plugin_doc(plugin, plugin_type, loader, search_paths):
         # if the plugin lives in a non-python file (eg, win_X.ps1), require the corresponding python file for docs
         result = loader.find_plugin_with_context(plugin, mod_type='.py', ignore_deprecated=True, check_aliases=True)
         if not result.resolved:
-            raise PluginNotFound('%s was not found in %s' % (plugin, search_paths))
+            if result.redirect_list and len(result.redirect_list) > 1:
+                # take the last one in the redirect list, we may have successfully jumped through N other redirects
+                target_module_name = result.redirect_list[-1]
```

Comment:
Would it beneficial to show the entire redirect list like `' -> '.join(result.redirect_list)`? I might be misunderstanding what `redirect_list` is at this point in time.

Reply:
I think we should cut off the first element, since that's the original name.
```suggestion
                target_module_name = ' -> '.join(result.redirect_list[1:])
```
results in:
```
[WARNING]: module docker_container was redirected to community.docker.docker_container -> foo.bar.doc, which was not found in:
```
Without the `1`, it results in:
```
[WARNING]: module docker_container was redirected to docker_container -> community.docker.docker_container -> foo.bar.doc, which was not found in:
```
(When backported to stable-2.10 the chain is more interesting, since it will include community.general.docker_container :-) )

Should it only be done for ansible-doc, or also for all other messages?

Reply:
I feel like it should be consistent and showing the full chain will definitely help debug problems in the future if they occur. I agree that `[1:]` is a better error message.

Reply:
Implemented in 345369997b5524970d8ca6b6ae751540219980ec.